File: Jack.Presentation/Controllers/API/TipoItemController.vb
Imports System.Net
Imports System.Web.Http

Namespace Controllers.API
    Public Class TipoItemController
        Inherits ApiController

        <HttpGet>
        Public Function GetValues() As IList(Of Model.TipoItem)

            Dim lstRetorno As List(Of Model.TipoItem) = Nothing
            Dim oBusiness As Business.TipoItem

            Try
                oBusiness = New Business.TipoItem()
                lstRetorno = oBusiness.LoadAll()

            Catch ex As Exception
                lstRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return lstRetorno

        End Function

        <HttpGet>
        Public Function GetValue(<FromUri> ID As Integer) As Model.TipoItem

            Dim oRetorno As Model.TipoItem
            Dim oBusiness As Business.TipoItem

            Try
                oBusiness = New Business.TipoItem()
                oRetorno = oBusiness.Find(ID)

            Catch ex As Exception
                oRetorno = Nothing
            Finally
                oBusiness = Nothing
            End Try

            Return oRetorno

        End Function

        <HttpPost>
        Public Sub Salvar(<FromUri> oFamily As Model.TipoItem)

            Dim oBusiness As Business.TipoItem
            Try
                oBusiness = New Business.TipoItem()
                oBusiness.Update(oFamily)
            Finally
                oBusiness = Nothing
            End Try

        End Sub

        <HttpDelete>
            Public Sub Delete(<FromUri> ID As Integer)
            Dim oBusiness As Business.TipoItem
            Dim oDelete As Model.TipoItem
            Try
                oBusiness = New Business.TipoItem()
                oDelete = New Model.TipoItem()
                oDelete.Codigo = ID
                    oBusiness.Delete(oDelete)

                Finally
                    oBusiness = Nothing
                End Try

            End Sub
        End Class
End Namespace

File: Jack.Presentation.Web/Controllers/API/TipoItemController.cs
using System;
using System.Collections.Generic;
using System.Web.Http;
using Application = Jack.Application;
using Model = Jack.Model;


namespace Controllers.API
{
    public class TipoItemController : ApiController
	{

		[HttpGet()]
		public IList<Model.TipoItem> GetValues()
		{

			IList<Model.TipoItem> lstRetorno = null;
			Application.TipoItem oApplication = default(Application.TipoItem);

            try
            {
                oApplication = new Application.TipoItem();
                lstRetorno = oApplication.LoadAll();

            }
            catch (Exception ex)
            {
                lstRetorno = null;
                throw ex;
            }
            finally
            {
                oApplication = null;
            }

			return lstRetorno;

		}

		[HttpGet()]
		public Model.TipoItem GetValue( [FromUri()] int ID)
		{

			Model.TipoItem oRetorno = default(Model.TipoItem);
			Application.TipoItem oApplication = default(Application.TipoItem);

			try {
				oApplication = new Application.TipoItem();
				oRetorno = oApplication.Find(ID);

			}
            catch (Exception ex)
            {
				oRetorno = null;
                throw ex;
            }
            finally
            {
				oApplication = null;
			}

			return oRetorno;

		}

		[HttpPost()]
		public void Salvar( [FromUri()] Model.TipoItem oFamily)
		{
			Application.TipoItem oApplication = default(Application.TipoItem);
			try
            {
				oApplication = new Application.TipoItem();
				oApplication.Update(oFamily);
			}
            finally
            {
				oApplication = null;
			}

		}

		[HttpDelete()]
		public void Delete(	[FromUri()] int ID)
		{
			Application.TipoItem oApplication = default(Application.TipoItem);
			Model.TipoItem oDelete = default(Model.TipoItem);
			try {
				oApplication = new Application.TipoItem();
				oDelete = new Model.TipoItem();
				oDelete.Codigo = ID;
				oApplication.Delete(oDelete);

			}
            finally
            {
				oApplication = null;
			}

		}
	}
}
